<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Data Preparation &#8212; DemuxNet 1.1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=fc837d61"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Usage" href="usage.html" />
    <link rel="prev" title="Installation" href="installation.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="data-preparation">
<h1>Data Preparation<a class="headerlink" href="#data-preparation" title="Link to this heading">¶</a></h1>
<p>DemuxNet requires a sparse single-cell expression matrix in <strong>RDS</strong> format
(genes as rows, cells as columns).</p>
<section id="recommendation">
<h2>Recommendation<a class="headerlink" href="#recommendation" title="Link to this heading">¶</a></h2>
<p>Before demultiplexing, filter the cell matrix to retain only high-confidence cells.
You can refer to the <code class="docutils literal notranslate"><span class="pre">assignment_confidence_table.csv</span></code> generated by 10x Cellranger multi:</p>
<p><code class="docutils literal notranslate"><span class="pre">your/path/to/outs/multi/multiplexing_analysis/assignment_confidence_table.csv</span></code></p>
</section>
<section id="r-preprocessing-script">
<h2>R Preprocessing Script<a class="headerlink" href="#r-preprocessing-script" title="Link to this heading">¶</a></h2>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">deMULTIplex2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>

<span class="c1"># ======= Load Files =======</span>
<span class="n">cmo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">Read10X</span><span class="p">(</span><span class="s">&#39;/your/path/to/outs/multi/count/raw_feature_bc_matrix/&#39;</span><span class="p">)</span>
<span class="n">cmo_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">&#39;your/path/to/outs/multi/multiplexing_analysis/assignment_confidence_table.csv&#39;</span><span class="p">)</span>

<span class="c1"># ======= Data Preprocessing ========</span>
<span class="c1"># Filter out Blank and Multiplet assignments</span>
<span class="n">filered_cmo_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cmo_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">Assignment</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Blank&#39;</span><span class="p">,</span><span class="s">&#39;Multiplet&#39;</span><span class="p">)))</span>

<span class="c1"># Extract cell-gene matrix</span>
<span class="n">cmo_mtx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cmo</span><span class="o">$</span><span class="n">`Gene Expression`</span><span class="p">[,</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">cmo</span><span class="o">$</span><span class="n">`Gene Expression`</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">filered_cmo_df</span><span class="o">$</span><span class="n">Barcode</span><span class="p">]</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">cmo_mtx</span><span class="p">))</span>

<span class="c1"># Extract cell-CMO tag information</span>
<span class="n">cmo_tag_mtx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">cmo</span><span class="o">$</span><span class="n">`Multiplexing Capture`</span><span class="p">[,</span><span class="w"> </span><span class="n">filered_cmo_df</span><span class="o">$</span><span class="n">Barcode</span><span class="p">])</span>

<span class="c1"># ======= Tagging using Traditional Methods =======</span>
<span class="n">cmo_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">demultiplexTags</span><span class="p">(</span><span class="n">cmo_tag_mtx</span><span class="p">,</span><span class="w"> </span><span class="n">plot.path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;./&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">plot.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;cmo&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">plot.diagnostics</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="n">cmo_assign_table</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cmo_res</span><span class="o">$</span><span class="n">assign_table</span>
<span class="n">cmo_assign_table</span><span class="o">$</span><span class="n">Barcode</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">cmo_assign_table</span><span class="p">)</span>

<span class="c1"># Create a dataframe with labeled barcodes</span>
<span class="n">cmo_bc_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">Barcode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">cmo_mtx</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">left_join</span><span class="p">(</span><span class="n">cmo_assign_table</span><span class="p">[,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Barcode&quot;</span><span class="p">,</span><span class="s">&quot;final_assign&quot;</span><span class="p">)],</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Barcode&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">barcode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">final_assign</span><span class="p">,</span><span class="w"> </span><span class="n">Barcode</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;_&#39;</span><span class="p">))</span>

<span class="nf">print</span><span class="p">(</span><span class="nf">identical</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">cmo_mtx</span><span class="p">),</span><span class="w"> </span><span class="n">cmo_bc_df</span><span class="o">$</span><span class="n">Barcode</span><span class="p">))</span>

<span class="c1"># Replace with labeled Barcodes</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">cmo_mtx</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cmo_bc_df</span><span class="o">$</span><span class="n">barcode</span>

<span class="c1"># ====== Save Data =========</span>
<span class="nf">saveRDS</span><span class="p">(</span><span class="n">cmo_mtx</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;your/path/to/cmo_mtx.rds&#39;</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="example-matrix-format">
<h2>Example Matrix Format<a class="headerlink" href="#example-matrix-format" title="Link to this heading">¶</a></h2>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>                   CMO303_AAACCCAAGCACTCCG-1 CMO301_AAACCCAAGCGATGCA-1
ENSONIG00000018423                     .                          .
ENSONIG00000032772                     .                          .
ENSONIG00000007536                     .                          .
ENSONIG00000033498                     .                          .
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper"><div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="installation.html" title="previous chapter">Installation</a></li>
      <li>Next: <a href="usage.html" title="next chapter">Usage</a></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2026, Hu lab.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/data_preprocessing.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>